A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/A320_Neo_EICAS.js

```diff
@@ -82,22 +95,52 @@ class A320_Neo_EICAS extends Airliners.BaseEICAS {
             SimVar.SetSimVarValue("L:ACPowerAvailable","bool",0);
         }
 
-        // Check if engine is on so self test doesn't appear when not starting from cold and dark
-        if (engineOn) {
-            this.selfTestDiv.style.display = "none";
-            this.selfTestTimerStarted = true;
-        }
-        // Check if external power is on & timer not already started
-        if ((externalPower || apuOn) && !this.selfTestTimerStarted) {
-            this.selfTestTimer = 14.25;
-            this.selfTestTimerStarted = true;
-        } // timer
-        if (this.selfTestTimer >= 0) {
-            this.selfTestTimer -= _deltaTime / 1000;
-            if (this.selfTestTimer <= 0) {
-                this.selfTestDiv.style.display = "none";
+        /**
+         * Self test on top ECAM screen
+         **/
+        
+        let topSelfTestCurrentKnobValue = SimVar.GetSimVarValue("LIGHT POTENTIOMETER:22", "number");
+        
+        if(((topSelfTestCurrentKnobValue >= 0.1 && this.topSelfTestLastKnobValue < 0.1) || (externalPowerChanged || apuStatusChanged)) && !this.topSelfTestTimerStarted) {
+            this.topSelfTestDiv.style.display = "block";
+            this.topSelfTestTimer = 14.25;
+            this.topSelfTestTimerStarted = true;
+        }
+        
```

Comment:
I think this is causing PFD and MFD to go into self test when turning on the APU, even though I'm on external power.

Also causes self test when leaving external power, even though my apu or engine is up and running.

Reply:
https://github.com/flybywiresim/a32nx/issues/402